src/burnchains/bitcoin/indexer.rs

```diff
@@ -600,17 +696,102 @@ impl BitcoinIndexer {
             // try again
             start_block = start_block.saturating_sub(REORG_BATCH_SIZE);
             reorg_spv_client =
-                self.setup_reorg_headers(&orig_spv_client, reorg_headers_path, start_block)?;
+                self.setup_reorg_headers(&orig_spv_client, reorg_headers_path, start_block, false)?;
         }
 
-        debug!("Bitcoin headers history is consistent up to {}", new_tip);
+        if check_chain_work {
+            let reorg_total_work = reorg_spv_client.update_chain_work()?;
+            let orig_total_work = orig_spv_client.get_chain_work()?;
+
+            debug!("Bitcoin headers history is consistent up to {}. Orig chainwork: {}, reorg chainwork: {}", new_tip, orig_total_work, reorg_total_work);
+
+            if orig_total_work < reorg_total_work {
+                let reorg_tip = reorg_spv_client.get_headers_height()?;
+                let hdr_reorg = reorg_spv_client
+                    .read_block_header(reorg_tip - 1)?
+                    .expect("FATAL: no tip hash for existing chain tip");
+                info!(
+                    "New canonical Bitcoin chain found! New tip is {}",
+                    &hdr_reorg.header.bitcoin_hash()
+                );
```

Comment:
Does this log line appear on every new bitcoin block, or only on reorgs?

Reply:
Only on reorgs.  This can only get hit if the newly-discovered headers (the "reorg" headers) have more total work than the headers currently in the headers DB (the "orig" headers).